### ISSUE TITLE
False positive for CA2225 when a base class already has the method defined

### ISSUE THREAD

**Issue by matt-edmondson** [dotnet/roslyn-analyzers#6640]
### Analyzer

**Diagnostic ID**: [CA2225](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2225): `Operator overloads have named alternates`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 7 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.302](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

CA2225 is triggered even though the requested method is defined and publicly available on the base class.

### Steps To Reproduce

```
namespace ktsu.io
{
	public class A
	{
		public bool ToBoolean() => this is not null;
	}

	public class B : A
	{
		public static implicit operator bool(B value) => value?.ToBoolean() ?? false;
	}
}
```
### Expected behavior
`A.ToBoolean()` should be found by the analyzer as a candidate method.

### Actual behavior
`A.ToBoolean()` is not considered when checking if B has an implementation for the requested methods

**Comment by NewellClark** [comment]
I would like to be assigned to this issue. Should be an easy fix.
